Common/util/string_utils.cpp

```diff
@@ -203,5 +204,12 @@ void StrUtil::WriteString(const String &s, Stream *out)
         out->Write(s.GetCStr(), len);
 }
 
+void StrUtil::WriteCStr(const String &s, Stream *out, int max_len)
+{
+    max_len = Math::Min(max_len, s.GetLength());
```

Comment:
hah, didn't even notice that.  Clearly more tests needed :)  Returning the written size (ignoring null char so you can compare with strlen) would be a good start I think.